network.go

```diff
@@ -438,11 +438,9 @@ func (n *network) updateSvcRecord(ep *endpoint, isAdd bool) {
 	var epList []*endpoint
 	n.WalkEndpoints(func(e Endpoint) bool {
 		cEp := e.(*endpoint)
-		cEp.Lock()
-		if cEp.container != nil {
+		if _, hasSandbox := cEp.getSandbox(); hasSandbox {
```

Comment:
It's not that bad :-). You just need to move those methods to Sandbox receiver and invoke them as Sandbox methods over here.